libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt

```diff
@@ -154,12 +155,19 @@ object MapboxNativeNavigatorImpl : MapboxNativeNavigator {
 
     // Routing
 
-    override suspend fun setPrimaryRoute(
-        routeWithStartingLeg: Pair<NavigationRoute, Int>?
-    ): RouteInfo? = suspendCancellableCoroutine { continuation ->
-        navigator!!.setPrimaryRoute(
-            routeWithStartingLeg?.first?.nativeRoute(),
-            routeWithStartingLeg?.second ?: 0
+    override suspend fun setRoutes(
+        primaryRoute: NavigationRoute?,
+        startingLeg: Int,
+        alternatives: List<NavigationRoute>,
+    ): Expected<String, SetRoutesResult> = suspendCancellableCoroutine { continuation ->
+        navigator!!.setRoutes(
+            primaryRoute?.let { route ->
+                SetRoutesParams(
+                    route.nativeRoute(),
+                    startingLeg,
+                    alternatives.map { it.nativeRoute() }
+                )
+            }
         ) { result ->
             result.onError {
                 logE(
```

Comment:
the error message must be changed: 
`"Failed to set the primary route with alternatives, an active guidance session will not function correctly. Reason: $it"`

Reply:
thank you, updated